src/core/osmLayer.js

```diff
@@ -441,23 +429,55 @@ geo.osmLayer = function (arg) {
     worldPt2[1] = Math.max(worldPt2[1], -180.0);
     worldPt2[1] = Math.min(worldPt2[1], 180.0);
 
+    /// Compute tile zoom
+    worldDelta = Math.abs(worldPt2[0] - worldPt1[0]);
+    worldDeltaY = Math.abs(worldPt2[1] - worldPt1[1]);
+
+    displayDeltaX = urx - llx;
+    displayDeltaY = lly - ury;
+
+    /// Reuse variables
+    if (displayDeltaY > displayDeltaX) {
+      displayDeltaX = displayDeltaY;
+      worldDelta = worldDeltaY;
+    }
+
+    noOfTilesRequired = Math.pow(2, Math.floor(
+      Math.log2(Math.round(displayDeltaX / 256.0))));
+    worldDeltaPerTile = worldDelta / noOfTilesRequired;
+
+    /// Minimize per pixel distortion
+    for (i = 20; i > 2; i = i - 1) {
+      if (!minDistWorldDeltaPerTile ||
+            Math.abs(360.0 / Math.pow(2, i) - worldDeltaPerTile) <
+           minDistWorldDeltaPerTile) {
+        minDistWorldDeltaPerTile =
+          Math.abs(360.0 / Math.pow(2, i) - worldDeltaPerTile);
+        m_zoom = i;
+      }
+    }
+
+    if (m_zoom < 2) {
```

Comment:
Right, that is intentional. Low level zoom levels are almost useless as they are too pixelated. Also, their at 0 and 1 (as well), the size of the texture is not 256 (but I have to verify again).